external/source/exploits/CVE-2022-34918/src/util.c

```diff
@@ -0,0 +1,198 @@
+#define _GNU_SOURCE
+#include <stdint.h>
+#include <stdlib.h>
+#include <sched.h>
+#include <stdio.h>
+#include <errno.h>
+#include <fcntl.h>
+#include <string.h>
+#include <sys/utsname.h>
+
+#include "log.h"
+#include "util.h"
+
+struct kernel_info kernels[] = {
+/*
+ * The structure is declared as:
+ * struct kernel_info {
+ *      const char* kernel_version;
+ *      uint64_t io_ring_ctx_ref_free;
+ *      uint64_t io_rsrc_node_ref_zero;
+ *      uint64_t modprobe_path;
+ * };
+ * The last three fields are the offsets of the corresponding symbols
+ */
+// 22.04 LTS
```

Comment:
I'm not passionate about changing it now that it is here and I recognize this is an artifact of the original tool, but normally I'm a fan of placing the offsets in the ruby code and passing them in as arguments.  It makes expanding the target sets easier, and allows a forced exploit to run.  Again, I'm not super passionate about changing it, but in the future I think that's the better paradigm.